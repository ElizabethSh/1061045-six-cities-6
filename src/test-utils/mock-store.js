import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { createAPI } from "../services/api";

export const api = createAPI(() => {});
export const mockStore = (state) => {
  const configuredMockStore = configureMockStore([
    thunk.withExtraArgument(api),
  ]);
  return configuredMockStore(state);
};
