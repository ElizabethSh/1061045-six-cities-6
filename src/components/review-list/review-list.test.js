import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import ReviewList from "./review-list";
import { mockStore } from "../../test-utils/mock-store";

const reviews = [
  {
    offerId: 1,
    comment: `Good`,
    date: `2019-09-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: 4,
      isPro: false,
      name: `Max`,
    },
  },
  {
    offerId: 1,
    comment: `Cozy`,
    date: `2020-06-08T14:13:56.569Z`,
    id: 2,
    rating: 4,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: 5,
      isPro: true,
      name: `Alex`,
    },
  },
  {
    offerId: 2,
    comment: `Perfect`,
    date: `2020-05-28T14:13:56.569Z`,
    id: 3,
    rating: 5,
    user: {
      avatarUrl: `img/avatar-max.jpg`,
      id: 6,
      isPro: true,
      name: `Frederik`,
    },
  },
];

describe("ReviewList", () => {
  it("renders all reviews with correct content", () => {
    const store = mockStore({
      REVIEW: { placeReviews: reviews, isReviewsLoaded: true },
    });

    render(
      <Provider store={store}>
        <ReviewList />
      </Provider>
    );

    // Check that all comments are rendered
    expect(screen.getByText("Good")).toBeInTheDocument();
    expect(screen.getByText("Cozy")).toBeInTheDocument();
    expect(screen.getByText("Perfect")).toBeInTheDocument();

    // Check that all user names are rendered
    expect(screen.getByText("Max")).toBeInTheDocument();
    expect(screen.getByText("Alex")).toBeInTheDocument();
    expect(screen.getByText("Frederik")).toBeInTheDocument();

    // Check that Pro status is rendered for Pro users
    expect(screen.getAllByText("Pro")).toHaveLength(2);
  });

  it("renders empty list if there are no reviews", () => {
    const store = mockStore({
      REVIEW: { placeReviews: [], isReviewsLoaded: true },
    });

    render(
      <Provider store={store}>
        <ReviewList />
      </Provider>
    );

    // Should not find any review comments
    expect(screen.queryByText("Good")).not.toBeInTheDocument();
    expect(screen.queryByText("Cozy")).not.toBeInTheDocument();
    expect(screen.queryByText("Perfect")).not.toBeInTheDocument();
  });
});
