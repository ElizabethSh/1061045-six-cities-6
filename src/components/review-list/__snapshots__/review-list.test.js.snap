// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render correctly 1`] = `
<div>
  <ul
    class="reviews__list"
  >
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Alex
        </span>
        <span
          class="property__user-status"
        >
          Pro
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Cozy
        </p>
        <time
          class="reviews__time"
          datetime="2020-6-8"
        >
          June 2020
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Frederik
        </span>
        <span
          class="property__user-status"
        >
          Pro
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 100%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Perfect
        </p>
        <time
          class="reviews__time"
          datetime="2020-5-28"
        >
          May 2020
        </time>
      </div>
    </li>
    <li
      class="reviews__item"
    >
      <div
        class="reviews__user user"
      >
        <div
          class="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            class="reviews__avatar user__avatar"
            height="54"
            src="img/avatar-max.jpg"
            width="54"
          />
        </div>
        <span
          class="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        class="reviews__info"
      >
        <div
          class="reviews__rating rating"
        >
          <div
            class="reviews__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          class="reviews__text"
        >
          Good
        </p>
        <time
          class="reviews__time"
          datetime="2019-9-8"
        >
          September 2019
        </time>
      </div>
    </li>
  </ul>
</div>
`;
